test/unit/server/start.test.ts

```diff
@@ -0,0 +1,134 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * Lewis Daly <lewisd@crosslaketech.com>
+ --------------
+ ******/
+import { Server } from '@hapi/hapi'
+import { ServiceConfig } from '~/shared/config'
+
+const defaultMockConfig: ServiceConfig = {
+  PACKAGE: {
+    version: '11.0.0'
+  },
+  PORT: 1234,
+  HOST: 'auth-service.local',
+  ENDPOINT_CACHE_CONFIG: {
+    expiresIn: 5000,
+    generateTimeout: 5000
+  },
+  ENDPOINT_SERVICE_URL: 'central-ledger.local',
+  ERROR_HANDLING: {
+    includeCauseExtension: true,
+    truncateExtensions: true,
+  },
+  INSTRUMENTATION: {
+    METRICS: {
+      DISABLED: false,
+      labels: {
+        eventId: "*"
+      },
+      config: {
+        timeout: 5000,
+        prefix: "moja_3p_api"
+      }
+    }
+  }
+}
+
+let mockInitializeCache: any;
+let mockSetupMetrics: any;
+
+describe('start', () => {
+  beforeEach(async () => {
+    jest.resetAllMocks()
```

Comment:
I've seen in all beforeEach we are using jest.resetAllMocks() -> I belive we can configure it to be always done in jest.config.js file

Reply:
Yeah I've seen that setting as well - I'm hesitant to use it because (1) I don't want to lose the granularity and (2) I don't want too much _magic_ running our tests for us.

I'm happy to be convinced otherwise however...